Fixed app crash on tapping examples (Fixed #1480)

arundhati24 opened the pull request:
Fixes #1480 
Changes: Added null check for the five star rating bar. (Crash report can be seen in the linked issue)

iamareebjamal commented on app/src/main/java/org/fossasia/susi/ai/skills/skilldetails/SkillDetailsFragment.kt:
```diff
@@ -271,39 +271,41 @@ class SkillDetailsFragment : Fragment(), ISkillDetailsView {
 
-        tvFiveStarSkillRatingBar.visibility = View.VISIBLE
-        fiveStarSkillRatingBar.visibility = View.VISIBLE
+        if (fiveStarSkillRatingBar != null) {
```
Rating Bar should never be null

Check why it is instead of adding a null check

batbrain7 commented:
@arundhati24 this is an URGENT priority issue please work on this or else let me work on it ?

arundhati24 commented:
@batbrain7 @iamareebjamal I am unable to understand why the rating bar becomes null sometimes. It doesn't happen every time. It leads to a crash once in a few trials.
@batbrain7 I also removed those extra variables this time. Removed lateinit variables and used direct ids instead, even now I got the same crash once which says fiveStarRatingBar must not be null.

batbrain7 commented:
@arundhati24 could you help me reproduce the issue ?

batbrain7 commented:
@arundhati24 I found the issue , the issue is with the skills that are already rated, since it takes some time for the rating bar to fetch that whether the user has already rated this skill or not and if the skill is rated the stars are updated and then we can click on examples to use them normally.
Also, in the mean time in which the fetching of the number of stars is done in that period of time if we click on the examples the app crashes otherwise it works fine.
So I suggest you to add some progressbar to show the user that rating is being fetched and then let the user interact with the skill details fragment.
Also @iamareebjamal can we do anything better ?

batbrain7 commented:
Now it is a fairly easy job to do, please solve this quickly so that a new release can be made.

batbrain7 commented:
@iamareebjamal I think @arundhati24 is still working on this
